docs/source/install-script.rst

```diff
@@ -0,0 +1,108 @@
+:orphan: true
```

Comment:
I'm not completely sure what the purpose of this file is. Most of our users just want to know to run `python3 firedrake-install`. They aren't necessarily interested in the exact process followed by the installation. Personally I feel like the information you've added to this file should be moved to a couple of different places:
- The overview of the installation process should be added to a docstring for `firedrake-install`.
- The automatically generated list of dependencies could be used to improve [this part](https://firedrakeproject.org/download.html#system-requirements) of the website (possibly also including the PETSc configure options).
- Any other information, for example about `parallel_packages` and `wheel_blacklist`, could be added as additional commenting of `firedrake-install` (possibly refactoring at the same time).